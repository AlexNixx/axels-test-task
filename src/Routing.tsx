import { Route, Routes } from 'react-router-dom';
import { HomePage } from './pages';
import { PropertyPage } from './pages';
import { NotFoundPage } from './pages';

export const Routing = () => (
    <Routes>
        <Route path='/' element={<HomePage />} />
        <Route path='/property/:id' element={<PropertyPage />} />
        <Route path='*' element={<NotFoundPage />} />
    </Routes>
);
