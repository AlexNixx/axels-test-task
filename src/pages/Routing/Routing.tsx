import { Route, Routes } from 'react-router-dom';

import { HomePage } from '../HomePage/HomePage';
import { PropertyPage } from '../PropertyPage/PropertyPage';
import { NotFoundPage } from '../NotFoundPage/NotFoundPage';

export const Routing = () => (
    <Routes>
        <Route path='/' element={<HomePage />} />
        <Route path='/property/:id' element={<PropertyPage />} />
        <Route path='*' element={<NotFoundPage />} />
    </Routes>
);
