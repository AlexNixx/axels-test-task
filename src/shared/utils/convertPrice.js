import PropTypes from 'prop-types';

export const convertPrice = price =>
    new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: 'USD'
    }).format(price);

convertPrice.propTypes = {
    price: PropTypes.number
};
