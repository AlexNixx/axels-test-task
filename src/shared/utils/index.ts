export * from './test/renderWithProviders';
export * from './test/showLocation';
export * from './convertPrice/convertPrice';
