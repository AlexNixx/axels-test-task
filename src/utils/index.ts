export * from './test/renderWithProviders';
export * from './test/showLocation';
export * from './test/mockProperties';
export * from './convertPrice/convertPrice';
