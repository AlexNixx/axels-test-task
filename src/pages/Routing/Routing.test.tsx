import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Routing } from './Routing';

import { renderWithProviders, ShowLocation } from '../../utils';

describe('Test App Routing', () => {
    test('landing on Home Page', async () => {
        const homeRoute = '/';
        const { container } = renderWithProviders(
            <>
                <ShowLocation />
                <Routing />
            </>,
            { initialRoute: homeRoute }
        );

        expect(await screen.findByTestId('homePage')).toBeInTheDocument();
        expect(screen.getByTestId('locationDisplay')).toHaveTextContent(
            homeRoute
        );
        expect(container).toMatchSnapshot();
    });

    test('landing on Property Details page', async () => {
        const propertyRoute = '/property/2';
        const { container } = renderWithProviders(
            <>
                <ShowLocation />
                <Routing />
            </>,
            { initialRoute: propertyRoute }
        );

        expect(await screen.findByTestId('propertyPage')).toBeInTheDocument();
        expect(screen.getByTestId('locationDisplay')).toHaveTextContent(
            propertyRoute
        );
        expect(container).toMatchSnapshot();
    });

    test('landing on a bad page', async () => {
        const badRoute = '/some/bad/route';

        const { container } = renderWithProviders(
            <>
                <ShowLocation />
                <Routing />
            </>,
            { initialRoute: badRoute }
        );

        expect(screen.getByTestId('notFoundPage')).toBeInTheDocument();
        expect(screen.getByTestId('locationDisplay')).toHaveTextContent(
            badRoute
        );
        expect(container).toMatchSnapshot();
    });
});
