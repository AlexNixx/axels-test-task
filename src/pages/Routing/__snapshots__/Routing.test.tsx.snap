// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Routing landing on Home Page 1`] = `
<div>
  <div
    data-testid="locationDisplay"
  >
    /
  </div>
  <div
    class="sc-jSwlEQ ePyzur container"
    data-testid="homePage"
  >
    <div
      class="sc-gKHVLF eEIIgr row"
    >
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/1"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/2"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/3"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/4"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/5"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/6"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/7"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/8"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/9"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-iBAaJG clwkyP col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <a
          data-testid="propertyLink"
          href="/property/10"
        >
          <div
            class="sc-bcPKhP bZwkZu card"
            data-testid="propertyCard"
          >
            <img
              alt="property Cozy Apartment in the City Center"
              class="card-img-top sc-grXZZQ dsaGvy"
              src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
                data-testid="cardTitle"
              >
                Cozy Apartment in the City Center
              </div>
              <div
                class="card-subtitle h6"
                data-testid="cardPrice"
              >
                $1,200.00
              </div>
              <p
                class="card-text"
                data-testid="cardAddress"
              >
                123 Main Street, Kyiv, Ukraine
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test App Routing landing on Property Details page 1`] = `
<div>
  <div
    data-testid="locationDisplay"
  >
    /property/2
  </div>
  <div
    class="sc-dkjKgF hkMOcG container"
    data-testid="propertyPage"
  >
    <a
      href="/"
    >
      <button
        class="sc-eDnVMP hlhphY btn btn-dark"
        type="button"
      >
        Back
      </button>
    </a>
    <div
      class="row"
    >
      <div
        class="col-lg-7 col-12"
      >
        <div
          class="carousel slide"
        >
          <div
            class="carousel-indicators"
          >
            <button
              aria-current="true"
              aria-label="Slide 1"
              class="active"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 2"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 3"
              data-bs-target=""
              type="button"
            />
          </div>
          <div
            class="carousel-inner"
          >
            <div
              class="active carousel-item"
            >
              <img
                alt="Cozy Apartment in the City Center-photo"
                class="sc-hLclGa ccoZNc"
                src="https://media.istockphoto.com/id/1322575582/photo/exterior-view-of-modern-apartment-building-offering-luxury-rental-units-in-silicon-valley.jpg?s=170667a&w=0&k=20&c=aL9w8U5MHZHJnsxMo3n6Nqo1rKNQSkW-gy-qd5NA8X8="
              />
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Cozy Apartment in the City Center-photo"
                class="sc-hLclGa ccoZNc"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTt2QX3mmexpRDhi6LgigPXGLCrLp-ZuSfAi--kI9xURmgfBrKlW_zFQ8Y0IzAqNkAjA5Q&usqp=CAU"
              />
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Cozy Apartment in the City Center-photo"
                class="sc-hLclGa ccoZNc"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ87P2uJ3BtSk_RMFepl8YInCagx9togVrSW3q9Z8qs56CJn-oXQ1WhVPaE-Lmx23dnvtI&usqp=CAU"
              />
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="visually-hidden"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="visually-hidden"
            >
              Next
            </span>
          </a>
        </div>
      </div>
      <div
        class="col"
      >
        <h2
          data-testid="propertyTitle"
        >
          Cozy Apartment in the City Center
        </h2>
        <h5
          data-testid="propertySeller"
        >
          Seller: 
          John Doe
        </h5>
        <h6
          data-testid="propertyPrice"
        >
          Price: 
          $1,200.00
        </h6>
        <p
          data-testid="propertyDescription"
        >
          This cozy apartment is located in the heart of the city, close to all amenities and attractions. It features a spacious living room, a fully equipped kitchen, and a comfortable bedroom. The perfect place for your next stay in Kyiv!
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test App Routing landing on a bad page 1`] = `
<div>
  <div
    data-testid="locationDisplay"
  >
    /some/bad/route
  </div>
  <div
    data-testid="notFoundPage"
  >
    Not Found Page
  </div>
</div>
`;
